awx_collection/plugins/modules/tower_job_wait.py

```diff
@@ -126,6 +135,14 @@ def main():
     timeout = module.params.get('timeout')
     min_interval = module.params.get('min_interval')
     max_interval = module.params.get('max_interval')
+    interval = module.params.get('interval')
+
+    if min_interval is not None or max_interval is not None:
```

Comment:
would expand it to `(min_interval is not None or max_interval is not None ) and (not interval)`

if interval is given, then take that

Reply:
it might be valuable to still give a warning if min or max are set though, because this will break playbooks when we remove these parameters in a future release

Reply:
Yes, I think the ideal implementation is that the warning is a separate conditional - always warn `if min_interval is not None or max_interval is not None`. Then separate from that, use this math if `interval` is not given.

Reply:
The problem with `interval` is that we have a default on it, so it will never be none. And we need to have something set for interval (which is why it has a default). We could do `(min_interval is not None or max_interval is not None) and interval == 1` but we wouldn't know if the 1 was explicitly set or pulled in as the default.

The other option would be to remove the default and then if there is nothing set (min, max or interval) just make the interval = 1 internally.

Any preferences?

Reply:
yeah that is gross either way. If you did the `interval == 1` check, I'd be okay with that hack because it goes in the right direction.